plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1250,32 +1308,56 @@ public void finishDelete(ConnectorSession session, ConnectorTableHandle tableHan
             rowDelta.validateNoConflictingDataFiles();
         }
 
+        if (isUpdate) {
+            // Ensures a row that is updated by this commit was not deleted by a separate commit
```

Comment:
Use "Ensure" rather than "Ensures". The latter sounds weird since it is a shorthand for "The code below ensures"